dev/MRTCore/mrt/mrm/include/mrm/Checksums.h

```diff
@@ -0,0 +1,244 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#pragma once
+
+namespace Microsoft::Resources
+{
+
+struct DefChecksum
+{
+protected:
+    DEF_CHECKSUM m_cs;
+
+public:
+    typedef DEF_CHECKSUM Checksum;
+
+    DefChecksum() : m_cs(0) {}
+
+    Checksum GetChecksum() const { return m_cs; }
+
+    /*! 
+     * Adds a supplied data blob to a DEF checksum.
+     *
+     * Adding a data blob to a DEF checksum updates the partial
+     * checksum twice:
+     * - first, with a four-byte buffer containing the little-endian unsigned representation
+     *   of the length of the buffer, in bytes.
+     * - next, with the contents of the buffer
+     * 
+     * A non-NULL buffer of length 0 adds just the buffer length to the checksum as described above.
+     * A NULL buffer adds buffer length 0 followed by a four zero-bytes.
+     * 
+     * \param partialChecksum
+     * A partially calculated checksum.
+     * 
+     * \param data
+     * The buffer to be added to the checksum
+     *
+     * \param dataSizeInBytes
+     * The size of data, in bytes.
+     *
+     * \param status
+     * If present, reports extended status when an error occurs:
+     * - If status is non-NULL, reports INVALID_ARG if data is NULL and dataSizeInBytes is greater than 0.
+     *
+     * \return Checksum
+     * Returns the updated checksum
+     */
+    static Checksum ComputeChecksum(
+        _In_ Checksum partialChecksum,
+        _In_reads_bytes_opt_(dataSizeInBytes) const BYTE* data,
+        _In_ UINT32 dataSizeInBytes);
+
+    /*! 
```

Comment:
is this a doc format? or feeds a doc generation tool?

Reply:
The comments under mrt/mrm are inconsistent. Parts of the code base did at some point aspire to allow doc generation, but that was long abandoned and also no longer makes sense since these APIs are not exposed to anyone. We at some point need to do a pass over all the comments and turn them into the same style.

Reply:
that works for me, thanks.